// Copyright 2016, Xuanting Cai
// Email: caixuanting@gmail.com

syntax = "proto2";

import "utils/proto/time_interval.proto";

package interface;

enum SecurityType { 
	CTP_FUTURE = 1; 
}

// Next tag: 14.
message Security {
  optional string country = 1;
  optional string id = 2;
  optional string exchange = 3;
  optional string name = 4;
  optional string product = 5;
  optional SecurityType security_type = 6;
  optional int32 multiplier = 7;
  optional double tick = 8;
  optional double long_margin = 9;
  optional double short_margin = 10;
  optional bool is_trading = 12;
  optional string open_date = 13;
  repeated util.TimeInterval time_interval = 11;
}

message SecurityPool {
  repeated Security security = 1;
}